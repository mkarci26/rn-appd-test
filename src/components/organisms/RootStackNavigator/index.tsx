import * as React from 'react';
import { ReactElement } from 'react';

export type RootStackProps = {
  appPagesArray: Array<AppPage>;
};

export type AppPage = {
  screenName: keyof any;
  componentName: any;
};



export const RootStackNavigator = ({ appPagesArray }: RootStackProps): ReactElement => (
  <>Home Screen</>
);

export default RootStackNavigator;
