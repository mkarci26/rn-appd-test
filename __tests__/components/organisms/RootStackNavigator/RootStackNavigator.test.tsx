import 'react-native';
import React from 'react';
import { render } from '@testing-library/react-native';
import RootStackNavigator, { AppPage } from 'components/organisms/RootStackNavigator';

describe('Root Navigation Test Suite', () => {

  const EducationalOverlay = () => <>Some screen text</>;

  const appPagesMock: Array<AppPage> = [
    { screenName: 'SomeScreen', componentName: EducationalOverlay },
  ];

  it('renders correctly', () => {
    const { toJSON } = render(
      <RootStackNavigator appPagesArray={appPagesMock} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
