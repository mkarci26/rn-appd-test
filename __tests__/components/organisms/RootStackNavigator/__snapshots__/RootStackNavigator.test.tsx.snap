// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root Navigation Test Suite renders correctly 1`] = `"Home Screen"`;
